docs/tools/firmware.sh

```diff
@@ -132,59 +129,58 @@ create_rpm () {
 	fi
 
 	# Create the spec file
-	echo "Name:       apple-firmware" > $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "Version:    ${ver}" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "Release:    1" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "Summary:    Wi-Fi and Bluetooth firmware for T2 Macs" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "License:    Proprietary" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-
-	echo -e "\n%description" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "Wi-Fi and Bluetooth firmware for T2 Macs" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-
-	echo -e "\n%install" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "mkdir -p %{buildroot}/usr/lib/firmware/brcm" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "install -m 644 * %{buildroot}/usr/lib/firmware/brcm" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-
-	echo -e "\n%post" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "modprobe -r brcmfmac_wcc || true" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "modprobe -r brcmfmac || true" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "modprobe brcmfmac || true" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "modprobe -r hci_bcm4377 || true" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "modprobe hci_bcm4377 || true" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-
-	echo -e "\n%files" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-	echo "/usr/lib/firmware/brcm/*" >> $HOME/rpmbuild/SPECS/apple-firmware.spec
-
-	# Set OS to Linux
-	if [ -f "$HOME/.rpmrc" ]
-	then
-		echo -e "\nBacking up existing .rpmrc"
-		cp ${verbose} $HOME/.rpmrc $HOME/.rpmrc_orj
-	fi
-	echo "buildostranslate: Darwin: Linux" > $HOME/.rpmrc
+	cat <<- EOF > $HOME/rpmbuild/SPECS/apple-firmware.spec
+		Name:       apple-firmware
+		Version:    ${ver}
+		Release:    1
+		Summary:    Wi-Fi and Bluetooth firmware for T2 Macs
+		License:    Proprietary
+		BuildArch: noarch
```

Comment:
Why are we using noarch? Aren't t2 macs x86_64

Reply:
Just in case you are considering M1 Macs, the python script is modified to include firmware for T2 Macs only.

Reply:
All firmware is `noarch`, its for anything that is not compiled for specific arch.

Reply:
I agree, but just to maintain uniformity across other packages, maybe go for x86_64.

Reply:
Dnf will not strip binaries and things when using `noarch` rpms, so its faster/cleaner

Reply:
I'll change other packages to noarch then.